easybuild/easyconfigs/c/cryptography/cryptography-41.0.1-GCCcore-12.3.0.eb

```diff
@@ -0,0 +1,143 @@
+easyblock = 'CargoPythonPackage'
+
+name = 'cryptography'
+version = '41.0.1'
+
+homepage = 'https://github.com/pyca/cryptography'
+description = "cryptography is a package designed to expose cryptographic primitives and recipes to Python developers."
+
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+toolchainopts = {'pic': True}
+
+builddependencies = [
+    ('binutils', '2.40'),
+    ('pkgconf', '1.9.5'),
+    ('Rust', '1.70.0'),  # required for cryptography
+    ('hatchling', '1.18.0'),
+    ('setuptools-rust', '1.6.0'),
+]
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('pycparser', '2.21'),
```

Comment:
This dep on `pycparser` can be removed as `cryptography` does not have a direct dep on it. A dep on `cffi` should be sufficient.